core/src/Streamly/Internal/Data/Stream/Type.hs

```diff
@@ -173,7 +199,396 @@ instance Monad m => Applicative (Stream m) where
-LIST_INSTANCES(Stream)
 NFDATA1_INSTANCE(Stream)
-FOLDABLE_INSTANCE(Stream)
-TRAVERSABLE_INSTANCE(Stream)
+
+-- XXX Renamed to foldWith because Stream has a Foldable instance having
+-- method fold.
+--
+-- | Fold a stream using the supplied left 'Fold' and reducing the resulting
+-- expression strictly at each step. The behavior is similar to 'foldl''. A
+-- 'Fold' can terminate early without consuming the full stream. See the
+-- documentation of individual 'Fold's for termination behavior.
+--
+-- >>> Stream.foldWith Fold.sum (Stream.enumerateFromTo 1 100)
+-- 5050
+--
+-- Folds never fail, therefore, they produce a default value even when no input
+-- is provided. It means we can always fold an empty stream and get a valid
+-- result.  For example:
+--
+-- >>> Stream.foldWith Fold.sum Stream.nil
+-- 0
+--
+-- However, 'foldMany' on an empty stream results in an empty stream.
+-- Therefore, @Stream.foldWith f@ is not the same as @Stream.head . Stream.foldMany
+-- f@.
+--
+-- @foldWith f = Stream.parse (Parser.fromFold f)@
+--
+-- /Pre-release/
+{-# INLINE foldWith #-}
+foldWith :: Monad m => Fold m a b -> Stream m a -> m b
+foldWith fld m = D.fold fld $ toStreamD m
+
+-- XXX Renamed to "list" because fromList is present in IsList instance.
+--
+-- |
+-- @
+-- fromList = 'Prelude.foldr' 'K.cons' 'K.nil'
+-- @
+--
+-- Construct a stream from a list of pure values. This is more efficient than
+-- 'K.fromFoldable' for serial streams.
+--
+-- @since 0.4.0
+{-# INLINE_EARLY list #-}
+list :: Monad m => [a] -> Stream m a
+list = fromStreamD . D.fromList
+
+------------------------------------------------------------------------------
+-- Conversion to and from direct style stream
+------------------------------------------------------------------------------
+
+-- These definitions are dependent on what is imported as S
+{-# INLINE fromStreamS #-}
+fromStreamS :: (Monad m) => S.Stream m a -> Stream m a
+fromStreamS = Stream . S.toStreamK
+
+{-# INLINE toStreamS #-}
+toStreamS :: (Monad m) => Stream m a -> S.Stream m a
+toStreamS = S.fromStreamK . getStream
+
+------------------------------------------------------------------------------
+-- Construction
+------------------------------------------------------------------------------
+
+-- | Build a stream by unfolding a /monadic/ step function starting from a
+-- seed.  The step function returns the next element in the stream and the next
+-- seed value. When it is done it returns 'Nothing' and the stream ends. For
+-- example,
+--
+-- @
+-- let f b =
+--         if b > 3
+--         then return Nothing
+--         else print b >> return (Just (b, b + 1))
+-- in drain $ unfoldrM f 0
+-- @
+-- @
+--  0
+--  1
+--  2
+--  3
+-- @
+--
+-- /Pre-release/
+--
+{-# INLINE toStreamD #-}
+toStreamD :: Applicative m => Stream m a -> D.Stream m a
+toStreamD (Stream m) = D.fromStreamK m
+
+{-# INLINE fromStreamD #-}
+fromStreamD :: Monad m => D.Stream m a -> Stream m a
+fromStreamD m = Stream $ D.toStreamK m
```

Comment:
These are already defined in master branch, these have been removed and added at a different place here.